Cargo.toml

```diff
@@ -19,3 +19,7 @@ openssl = "0.10"
 [dependencies.serde]
 version = "1.0"
 features = ["derive"]
+
+[features]
+default = ["key_openssl_pkey"]
+key_openssl_pkey = []
```

Comment:
Can you make openssl an optional dependency?

Reply:
I think this would make more sense when we add a secondary crypto provider such as ring.

Reply:
The traits as proposed in this PR can immediately be implemented by external crates, so I don't see why the OpenSSL dependency should be required.

Reply:
So, `openssl` itself is still needed for encryption (which I did not abstract in this PR), and we still use it for things like computing digests before passing them to the crypto layer.

Reply:
I have submitted #28, which makes openssl a fully optional dependency (well, it is abstracted away, right now it's the only crypto implementation and you'll need one).